Library/Homebrew/extend/os/linux/diagnostic.rb

```diff
@@ -139,6 +139,34 @@ def check_linuxbrew_bottle_domain
           e.g. by using homebrew instead).
         EOS
       end
+
+      def check_gcc_dependent_linkage
+        gcc_dependents = Formula.installed.select do |formula|
+          next false unless formula.tap&.core_tap?
+
+          formula.recursive_dependencies.map(&:name).include? "gcc"
+        rescue TapFormulaUnavailableError
+          false
+        end
+        return if gcc_dependents.empty?
+
+        badly_linked = gcc_dependents.select do |dependent|
+          keg = Keg.new(dependent.prefix)
+          keg.binary_executable_or_library_files.any? do |binary|
+            paths = binary.rpath.split(":")
+            versioned_linkage = paths.any? { |path| path.match?(%r{lib/gcc/\d+$}) }
+            unversioned_linkage = paths.any? { |path| path.match?(%r{lib/gcc/current$}) }
+
+            versioned_linkage && !unversioned_linkage
```

Comment:
What I should be doing here is checking that the `current` directory precedes the versioned directory whenever the versioned directory is found, but I'm still thinking of the right way to do that.

In any case, if this check is something we want to do, then this is still an improvement over not checking, and we can iterate on this later. But, suggestions very much welcome.